GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs

```diff
@@ -1152,11 +1116,14 @@ private void UpdatePlaceholders()
                 int deleteFolderPlaceholderAttempted = 0;
                 using (BlobSizes.BlobSizesConnection blobSizesConnection = this.blobSizes.CreateConnection())
                 {
-                    // A hash of the folder placeholders is only required if the platform expands directories
-                    HashSet<string> folderPlaceholders =
-                        GVFSPlatform.Instance.KernelDriver.EnumerationExpandsDirectories ?
-                        new HashSet<string>(placeholderFoldersListCopy.Select(x => x.Path), StringComparer.OrdinalIgnoreCase) :
-                        null;
+                    // A hash of the placeholders is only required if the platform expands directories
+                    HashSet<string> existingPlaceholders = null;
+                    if (GVFSPlatform.Instance.KernelDriver.EnumerationExpandsDirectories)
+                    {
+                        IEnumerable<string> allPlaceholders = placeholderFoldersListCopy.Select(x => x.Path)
+                            .Union(this.placeholderDatabase.GetAllFilePaths());

```

Comment:
nit: Please line up the `.Union()` with the `.Select()` above. This can also be done by putting `.Select()` on its own line.